From adb6e0fb99d251e027cda875105d8e16d024dd46 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Mon, 31 Dec 2018 09:31:33 +0100
Subject: [PATCH 0965/1200] arm: dts: a3720-db: add SD slot power supply
 regulator description

SD slot on Armada 3720 DB is supplied by fixed 3.3V regulator.
Add appropriate description in the DT. This patch fixes problems
with the card discovery, in case the power supply was disabled
by the OS.

Change-Id: I1bd3448cc9237f84ee219d0f08eb19a8893d9d21
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/62316
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1869
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 arch/arm/dts/armada-3720-db.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/dts/armada-3720-db.dts b/arch/arm/dts/armada-3720-db.dts
index 37053cb3ee..e5cb132d0c 100644
--- a/arch/arm/dts/armada-3720-db.dts
+++ b/arch/arm/dts/armada-3720-db.dts
@@ -97,6 +97,15 @@
 				  3300000 0x0>;
 		gpios = <&gpiosb 23 GPIO_ACTIVE_HIGH>;
 	};
+
+	vcc_sd_reg1: regulator@1 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_sd1";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		shutdown-delay-us = <1000000>;
+		gpio = <&expander0 4 GPIO_ACTIVE_HIGH>;
+	};
 };
 
 
@@ -149,6 +158,7 @@
 &sdhci0 {
 	bus-width = <4>;
 	vqmmc-supply = <&vcc_sd_reg0>;
+	vmmc-supply = <&vcc_sd_reg1>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdio_pins>;
 	status = "okay";
-- 
2.22.0

